app/controllers/ApplicationController.scala

```diff
@@ -47,6 +47,14 @@ trait ApplicationController extends BaseController with GatekeeperAuthWrapper {
       } yield Ok(applications(subApps, groupApisByStatus(apis)))
   }
 
+  def applicationPage(appId: String): Action[AnyContent] = requiresRole(Role.APIGatekeeper) {
+    implicit request => implicit hc =>
+      for {
```

Comment:
is there any reason to chain these in sequence? Could start the futures outside and then put them together after.

Reply:
Could do - will give it a go now.